enhancements/installer/single-node-installation-bootstrap-in-place.md

```diff
@@ -0,0 +1,694 @@
+   an image for deploying OpenShift on a single node.
+3. Boot the new image on the host.
+4. Bootstrap the deployment, generating a new master Ignition config
+   and the static pod definitions for OpenShift. Write them, along
+   with an RHCOS image, to the disk in the host.
+5. Reboot the host to the internal disk, discarding the ephemeral live
+   image environment and allowing the previously generated artifacts
+   to complete the installation and bring up OpenShift.
+
+### User Stories
+
+#### As an OpenShift user, I want to be able to deploy OpenShift on a supported single node configuration
+
+A user will be able to run the OpenShift installer to create a single-node
+deployment, with some limitations (see non-goals above). The user
+will not require special support exceptions to receive technical assistance
+for the features supported by the configuration.
+
+### Implementation Details/Notes/Constraints
+
+The installation images for single-node clusters will be unique for
+each cluster. The user or orchestration tool will create an
+installation image by combining the
+`bootstrap-in-place-for-live-iso.ign` created by the installer with an
+RHCOS live image using `coreos-install embed`. Making the image unique
+allows us to build on the existing RHCOS live image, instead of
+delivering a different base image, and means that the user does not
+need any infrastructure to serve Ignition configs to hosts during
+deployment.
+
+In order to add a viable, working etcd post reboot, we will take a
+snapshot of etcd and add it to the Ignition config for the host.
+After rebooting, we will use the restored `etcd-member` from the
+snapshot to rebuild the database. This allows etcd and the API service
+to come up on the host without having to re-apply all of the
+kubernetes operations run during bootstrapping.
+
+#### OpenShift-installer
+
+We will add a new `create single-node-ignition-config` command to the
+installer to create the `bootstrap-in-place-for-live-iso.ign` Ignition
+config.  This new target will *not* output `master.ign` and `worker.ign`
+files.
+
+The `bootstrap-in-place-for-live-iso.ign` Ignition config will have a
+different `bootkube.sh` from the default bootstrap Ignition. In
+addition to the standard rendering logic, the modified script will:
+
+1. Start `cluster-bootstrap` without required pods by setting `--required-pods=''`
+2. Run `cluster-bootstrap` with the `--bootstrap-in-place` option.
+3. Fetch the master Ignition and combine it with the original Ignition
+   config, the control plane static pod manifests, the required
+   kubernetes resources, and the bootstrap etcd database snapshot to
+   create a new Ignition config for the host.
+
+In order to successfully complete the installation, we must know
+where to write the operating system image on the host's local
+storage. Initially this information will be passed as a single value
+in the `install-config.yaml`, using a schema like
+
+```yaml
+coreOSInstallation:
```

Comment:
I approve of these changes. I would like to reserve the right to adjust the name of this field to bring it out of the implementation domain and into the user domain. I do not want to hold up merging this PR while we debate this name, however. We can work on this name as we work on adding additional fields, with the understanding that the name needs to be finalized soon (certainly before we ship 4.8).

Reply:
Thanks for noting.

IMO, I think that makes sense in general for all enhancements - we should bias towards merging and iterating on the details, once the fundamentals have been worked through. The fundamental here was the need for an API and its requirements, not the details of its design.